File: Array_to_Tree.cpp
TreeNode* arrayToBinaryTree(int arr[], int& index, int size) {
    if (index >= size || arr[index] == -1) {
        index++;
        return nullptr;
    }

    TreeNode* node = new TreeNode;
    node->data = arr[index++];
    node->left = arrayToBinaryTree(arr, index, size);
    node->right = arrayToBinaryTree(arr, index, size);
    return node;
}

void inOrderTraversal(TreeNode* root) {
    if (root) {
        inOrderTraversal(root->left);
        cout << root->data << " ";
        inOrderTraversal(root->right);
    }
}

File: Array_to_Tree.java
TreeNode arrayToBinaryTree(int[] arr, int[] index, int size) {
    if (index[0] >= size || arr[index[0]] == -1) {
        index[0]++;
        return null;
    }

    TreeNode node = new TreeNode(arr[index[0]++]);
    node.left = arrayToBinaryTree(arr, index, size);
    node.right = arrayToBinaryTree(arr, index, size);
    return node;
}

void inOrderTraversal(TreeNode root) {
    if (root != null) {
        inOrderTraversal(root.left);
        System.out.print(root.data + " ");
        inOrderTraversal(root.right);
    }
}
